docs/operations/problem-determination.md

```diff
@@ -0,0 +1,221 @@
+---
+sidebar_position: ?
+---
+
+# Problem Determination
+
+It is important to be able to investigate problems that occur in production.
+Team members have typically used one of these implementations:
+* existing Application Performance Management offering
+* custom solution leveraging avialable platform tools (for example,
+  those provided by a Cloud provider or OpenShift)
+
+Irrespective of the implementation, the experience of the team is that it is important
+to plan for problem determination up front and use metrics to identify when problems
+occur and need to be investigated. You can read more about the suggested approach
+for capturing metrics in the
+[metrics](https://github.com/nodeshift/nodejs-reference-architecture/blob/main/docs/operations/metrics.md)
+section.
+
+Once your metrics have identified that there is a problem, it will often fall into
+one of the common categories.
+
+The teams recommendations/guidance is based on the implementation and typical categories
+of problems that we have seen.
+
+## Recommended Components
+
+## Guidance
+
+### Common problems
+
+#### Memory Leak
+
+* Symptoms
+  * metrics report that application is slowly becoming less responsive 
+  * metrics report heap size is continuously growing despite stable load on the application
+  * metrics report increasing time spent running the garbage collector
+
+* Approach
+  * generate a sequence of [heap dumps](https://nodejs.org/dist/latest-v17.x/docs/api/cli.html#--heapsnapshot-signalsignal)
+    at 1 minute intervals, compare using chrome dev tools to see what is growing
+  * enable the [heap profiler](https://nodejs.org/dist/latest-v17.x/docs/api/cli.html#--heap-prof)
+    and look at where allocations are taking place
+
+Generating heap dumps will have a performance impact on the process both in terms of memory
+and execution when the dump is being generated. Some of the team's suggestions for limiting the impact:
+* only enable heap dumps for one process/instance of the application, leaving the others to better server customers
+* try to minimize the size of the heap dumps. For example, if you increased the memory allocated to the application
```

Comment:
Make sure to disable Kubernetes liveness checks (may require new app deployment with that setting) or the liveness check may kill your app before it finishes creating the heap dump.